File: 2-add-two-numbers/2-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        if(l1==null) return l2;
        if(l2==null) return l1;
        
        int carry=0;
        int res=0;
        int add=0;
        ListNode head=new ListNode();
        ListNode h=head;
            
        while(l1!=null || l2!=null){
            add= ((l1!=null)? l1.val:0 )+((l2!=null)? l2.val:0 )+carry;
            carry=add/10;
            ListNode t=new ListNode(add%10);
            h.next=t;
            h=t;
            if(l1!=null) l1=l1.next;
            if(l2!=null) l2=l2.next;
        }
        if(carry==1) {
            ListNode t=new ListNode(1);
            h.next=t;
        }
        
        return head.next;
    }
}

File: 2-add-two-numbers/2-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]: 
        start = ListNode()
        head=start
        carry=0
        while l1 or l2 or carry:
            sm=(l1.val if l1 else 0)+(l2.val if l2 else 0)+ carry
            carry=sm//10
            head.next=ListNode(sm%10)
            head=head.next
            if l1: l1=l1.next
            if l2: l2=l2.next
        return start.next
            